File: aspnet-in-practice/06/CS/CustomControls.ControlBuilders/Zone.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Web.UI;
using System.Collections;
using System.Web.UI.WebControls;

namespace CustomControls.ControlBuilders
{
	[ControlBuilder(typeof(ZoneBuilder))]
	[ParseChildren(false)]
	public class Zone : CMSPartBase
	{
		protected override void RenderContents(HtmlTextWriter writer)
		{
			for (int i = 0; i < this.Controls.Count; i++)
			{
				Controls[i].RenderControl(writer);
			}
		}

		protected override HtmlTextWriterTag TagKey
		{
			get
			{
				return HtmlTextWriterTag.Div;
			}
		}
	}

}

File: aspnet-in-practice/06/VB/CustomControls.Web/CustomControls.ControlBuilders/Zone.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports System.Web.UI
Imports System.Collections
Imports System.Web.UI.WebControls

<ControlBuilder(GetType(ZoneBuilder))>
 <ParseChildren(False)>
Public Class Zone
	Inherits CMSPartBase
	Protected Overrides Sub RenderContents(ByVal writer As HtmlTextWriter)
		For i As Integer = 0 To Me.Controls.Count - 1
			Controls(i).RenderControl(writer)
		Next
	End Sub

	Protected Overrides ReadOnly Property TagKey() As HtmlTextWriterTag
		Get
			Return HtmlTextWriterTag.Div
		End Get
	End Property
End Class